semgrep-core/src/analyzing/CFG_build.ml

```diff
@@ -44,6 +44,8 @@ type label_key = string * G.sid
  *)
 type state = {
   g : (F.node, F.edge) Ograph_extended.ograph_mutable;
+  (* We keep this so we can connect to unreachable nodes. *)
```

Comment:
Can you give an example of why we need enteri?
(like we did for exiti with 'return' as an example).

Reply:
Suppose we have a program like the following:
```
function foo () {
  return 2;
  
  function bar() {
    sink(source);
  }
}
```

It's my understanding that this is valid Javascript code.

If we try to analyse this program without using the enter node, then we have parts of the CFG which are unreachable by regular dataflow analysis. This means that taint analysis does not run on these functions. Right now, since Semgrep visits _all_ function nodes via `Visitor_AST`, this means we are losing functionality that Semgrep currently supports.

This easy fix (or hack) remedies that situation by simply making all nodes reachable in the CFG.

Reply:
Could be nice to add some of this explanation in the code, as a comment.